Fixes #27

harper-wavra opened the pull request:
Added Streamgages layer.
When you click on streamgage popup some basic information should appear:
-  Name and ID 
-  Station type
-  Lat/long
-  Most recent discharge value (says no data if none)
-  StreamStats gage page
-  NWIS


**Changes to Code**
map.component.html - displayed zoom level, lat and long on map (can remove later, I thought it was just nice to see for now)
map.component.ts:
- added on zoom, on mousemove, and on dragend event, in order to update streamgage layer
- subscribed to waterData and streamgages from mapservice
- Added addGeoJSON function, to add new streamgage layer to map
- Added updatePopup function, to update the streamgage popups with most recent data
- Added setBbox function, to set the bouding box when the map is changed.

config - added gagestats URLS
map.service.ts:
- added set and get streamgages, to get streamgages from gagestats services
- added set and get waterData, to get current discharge from waterservices


Created two new issues for [unit testing](https://github.com/USGS-WiM/StreamStats-National/issues/64), and plotting [hydrograph from NWIS](https://github.com/USGS-WiM/StreamStats-National/issues/65) (which Pete said was a level 2)

amedenblik commented on src/app/map/map.component.html:
```diff
@@ -1 +1,6 @@
-<div id="map"></div>
\ No newline at end of file
+<div id="map">
+    <div class="leaflet-bottom leaflet-left">
```
The team decided earlier on that we'll just include map scale in this bottom left area, at least for now. That's included in issue #10, so you can omit this for now.

amedenblik commented on src/app/map/map.component.ts:
```diff
@@ -1,19 +1,30 @@
-
-  constructor(private _mapService: MapService) { }
+  public streamgageLayer: any;
+  public currentLat: number = 41.1;
+  public currentLng: number = -98.7;
```
Can omit currentLat and currentLng

amedenblik commented on src/app/map/map.component.html:
```diff
@@ -1 +1,6 @@
-<div id="map"></div>
\ No newline at end of file
+<div id="map">
+    <div class="leaflet-bottom leaflet-left">
+        <span>Zoom Level: {{currentZoom}}</span> <br />
+        <span>Lat: {{currentLat.toFixed(4)}}, Lng: {{currentLng.toFixed(4)}} </span>
+    </div>
```
The team decided earlier on that we'll just include map scale in this bottom left area, at least for now. That's included in issue #10, so you can omit this for now. Looks good though!

amedenblik reviewed (changes requested):
Looks great! See a few small comments for requested changes.

I spot-checked a number of gages, and they look good. I did see one gage that had an invalid NWIS page: "Big Browns Creek near Union (021564301)" (it's east of Union, SC). [NWIS page](https://nwis.waterdata.usgs.gov/nwis/inventory/?site_no=021564301) and [StreamStats Gage Page](https://streamstatsags.cr.usgs.gov/gagepages/html/021564301.htm). This is also an error in the current StreamStats client. I wonder if this signifies an issue with GageStatsServices, or is this expected?

harper-wavra commented:
Removed currentlat and currentlong code. Removed the display of currentzoom, currentlat and currentlong from the map.

As far as that weird gage, it is just an entry in the station table in the gagestats DB. If it is not a real station it would just need to deleted from that table.
[Database Entry](https://user-images.githubusercontent.com/59063181/131677331-904f307c-db85-48fd-b11e-901b7ad334da.png)
